In [ ]:
# 📈 Statistical Summary

Notebook ini melakukan analisis statistik deskriptif seperti:
- Mean (rata-rata)
- Median (nilai tengah)
- Modus (nilai paling sering)
- Standard Deviation (penyebaran data)

Statistik ini penting untuk memahami seberapa tersebar dan terpusat data numerik yang dianalisis.

🎯 Tujuan:

Notebook ini akan melakukan analisis statistik dasar (statistik deskriptif) terhadap data numerik menggunakan:
-numpy
-statistics (modul bawaan Python)

In [2]:
import numpy as np
import statistics as stats
import pandas as pd

In [5]:
df = pd.read_csv('../data/nilai_siswa.csv')
df.head()

Nama  Nilai Kelas
0    Ali     80     A
1   Budi     90     B
2  Citra     78     A
3   Dina     75     C
4    Eko     85     B

In [6]:
nilai = df["Nilai"]

print("Mean (rata-rata):", np.mean(nilai))
print("Median (nilai tengah):", np.median(nilai))
print("Modus:", stats.mode(nilai))
print("Standar Deviasi:", np.std(nilai))


Mean (rata-rata): 81.6
Median (nilai tengah): 80.0
Modus: 80
Standar Deviasi: 5.314132102234569


🧠 Insight Sederhana
- Rata-rata nilai siswa adalah 81.6
- Nilai median: 80 → data relatif seimbang
- Nilai paling sering (modus): 75 (jika data duplikat ada)
- Penyebaran nilai cukup kecil, data terkonsentrasi